src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -98,16 +98,52 @@ public async Task<IEnumerable<ReviewModel>> GetRequestedReviews(string userName)
         }
 
         public async Task<(IEnumerable<ReviewModel> Reviews, int TotalCount, int TotalPages, int CurrentPage, int? PreviousPage, int? NextPage)> GetPagedReviewsAsync(
-            IEnumerable<string> search, IEnumerable<string> languages, bool? isClosed, IEnumerable<int> filterTypes, bool? isApproved, int offset, int limit, string orderBy)
+            IEnumerable<string> search, IEnumerable<string> languages, bool? isClosed, IEnumerable<int> filterTypes, bool? isApproved, bool notApprovedForFirstRelease, int offset, int limit, string orderBy)
         {
-            var result = await _reviewsRepository.GetReviewsAsync(search, languages, isClosed, filterTypes, isApproved, offset, limit, orderBy);
+            var result = await _reviewsRepository.GetReviewsAsync(
+                search: search, languages: languages, isClosed: isClosed,
+                filterTypes: filterTypes, isApproved: isApproved, 
+                notApprovedForFirstRelease: notApprovedForFirstRelease,
+                offset: offset, limit: limit, orderBy: orderBy);
+
+            List<ReviewModel> filteredResults = null;
+
+            // Fiter reviews based on if no other review with the same PackageName and Language in approved or approvedForFirstReleases
+            if (notApprovedForFirstRelease)
+            {
+                filteredResults = new List<ReviewModel>();
+                do
+                {
+                    foreach (var reviewGroup in result.Reviews.GroupBy(r => $"{r.PackageName}-{r.Language}", r => r))
+                    {
+                        var firstInGroup = reviewGroup.FirstOrDefault();
+                        if (firstInGroup != null)
+                        {
+                            var reviewsApprovedForFirstRelease = await _reviewsRepository.GetApprovedForFirstReleaseReviews(firstInGroup.Language, firstInGroup.PackageName);
+                            if (!reviewsApprovedForFirstRelease.Any())
+                            {
+                                filteredResults.AddRange(reviewGroup);
+                            }
+                        }
+                    }
+                    if (filteredResults.Count < limit)
+                    {
+                        result = await _reviewsRepository.GetReviewsAsync(
+                            search: search, languages: languages, isClosed: isClosed,
+                            filterTypes: filterTypes, isApproved: isApproved,
+                            notApprovedForFirstRelease: notApprovedForFirstRelease,
+                            offset: offset, limit: limit, orderBy: orderBy);
```

Comment:
Shouldn't offset be adjusted when querying for additional reviews to be packed in the filtered result set? Let's discuss about this. I am a bit confused how UI will fetch next set when next page is clicked.